dev-ml/cstruct/cstruct-6.2.0.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit dune multiprocessing
+
+DESCRIPTION="Map OCaml arrays onto C-like structs"
+HOMEPAGE="
+	https://github.com/mirage/ocaml-cstruct
+	https://opam.ocaml.org/packages/cstruct/
+"
+SRC_URI="https://github.com/mirage/ocaml-cstruct/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/ocaml-${P}"
+
+LICENSE="ISC"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="async lwt ocamlopt ppx sexp test unix"
+
+RDEPEND="
+	>=dev-lang/ocaml-4.08.0:=[ocamlopt?]
+	>=dev-ml/fmt-0.8.9:=
+
+	async? (
+		>=dev-ml/async-0.9.0:=
+		>=dev-ml/async_unix-0.9.0:=
+		>=dev-ml/core-0.9.0:=
+	)
+	lwt? ( dev-ml/lwt:= )
+	ppx? (
+		>=dev-ml/ppxlib-0.16.0:=
+		dev-ml/stdlib-shims:=
+	)
+	sexp? ( >=dev-ml/sexplib-0.9.0:= )
+"
+DEPEND="
+	${RDEPEND}
+	test? (
+		dev-ml/alcotest
+		dev-ml/crowbar
+		|| (
+			>=dev-lang/ocaml-5.1
+			>=dev-ml/ocaml-migrate-parsetree-2.1.0
+		)
+		dev-ml/ppx_sexp_conv
+		dev-ml/cppo
+		>=dev-ml/lwt-2.0.2
+	)
+"
+
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	ppx? ( sexp )
+	test? ( sexp ppx )
+"
+
+src_prepare() {
+	default
+
+	if has_version '>=dev-lang/ocaml-5.1'; then
+		eapply "${FILESDIR}/6.2-ocaml-5.1.patch"
+	fi
```

Comment:
ok, I think this might be unnecessary (as in: can always be applied, the dependency on ocaml-migrate-parsetree is redundant/unnecessary independent of ocaml version)

Reply:
I have literally no idea about this topic, nor ocaml in general, but seems like ocaml-migrate-parsetree has been superseded by ppxlib (https://ocaml.org/changelog/2023-10-23-omp-deprecation)?

Reply:
it still requires that individual libraries need to be ported. In this case it appears that the dependency on ocaml-migrate-parsetree is just an outdated artifact from such a port.